mlrun/execution.py

```diff
@@ -840,13 +848,16 @@ def update_artifact(self, artifact_object):
         """update an artifact object in the cache and the DB"""
         self._artifacts_manager.update_artifact(self, artifact_object)
 
-    def commit(self, message: str = "", completed=True):
+    def commit(self, message: str = "", completed=False):
         """save run state and optionally add a commit message
 
         :param message:   commit message to save in the run
         :param completed: mark run as completed
         """
-        completed = completed and self._state == "running"
+        # changing state to completed is allowed only when the execution is in running state
```

Comment:
```suggestion
        # changing state to completed(terminal state) is allowed only when the execution is in non-terminal state
```

Reply:
The move to `completion` is allowed only from `running`. There are many more terminal/non-terminal states that this does not apply to.  For example moving from `pending` to `completed` is not allowed but it is non-terminal to terminal so I don't think that specifying terminal / non-terminal here is relevant.
~~~
    @staticmethod
    def all():
        return [
            RunStates.completed,
            RunStates.error,
            RunStates.running,
            RunStates.created,
            RunStates.pending,
            RunStates.unknown,
            RunStates.aborted,
        ]

    @staticmethod
    def terminal_states():
        return [
            RunStates.completed,
            RunStates.error,
            RunStates.aborted,
        ]
~~~